### PR TITLE
feat: [CDS-36922]: env group consumption with gitops in pipeline

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
##### Summary:

Consume environment group entity with gitops in pipeline

##### Jira Links:

https://harness.atlassian.net/browse/CDS-36922

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructureWidget.tsx [hunk 1]
```diff
@@ -17,7 +17,10 @@ import {
   Dialog,
   Layout,
   MultiTypeInputType,
-  SelectOption
+  SelectOption,
+  SplitButton,
+  SplitButtonOption
+  // SelectWithSubmenu
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructureWidget.tsx:23]: remove this

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructureWidget.tsx [hunk 2]
```diff
@@ -213,24 +219,150 @@ export function DeployInfrastructureWidget({
...
+
+  const [environmentGroups, setEnvironmentGroups] = useState<EnvironmentGroupResponseDTO[]>()
+  const [environmentGroupsSelectOptions, setEnvironmentGroupsSelectOptions] = useState<SelectOption[]>()
+
+  useEffect(() => {
+    if (!environmentGroupsLoading && !environmentGroupsResponse?.data?.empty) {
+      const environmentGroupsList: EnvironmentGroupResponseDTO[] = []
+      if (environmentGroupsResponse?.data?.content) {
+        environmentGroupsResponse.data.content.forEach(environmentObj => {
+          environmentGroupsList.push({ ...environmentObj.envGroup })
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructureWidget.tsx:245]: You could have directly used Array.map here

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructureWidget.tsx [hunk 3]
```diff
@@ -420,16 +607,127 @@ export function DeployInfrastructureWidget({
...
+                          MultiTypeInputType.FIXED &&
+                          environmentsLoading)
+                      }
+                      placeholder={
+                        environmentsLoading
+                          ? getString('loading')
+                          : getString('cd.pipelineSteps.environmentTab.selectEnvironment')
+                      }
+                      multiTypeInputProps={{
+                        // onTypeChange: setEnvironmentRefType,
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructureWidget.tsx:664]: Remove this

File: src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructureWidget.tsx (hunk 2 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/DeployInfrastructureWidget.tsx:245]: 🤦🏻‍♂️ I'll update it

**Review by bhavya-sinha — DISMISSED** [review]

**Review by vkbansal — APPROVED** [review]

**Review by YounixIO — APPROVED** [review]

**Pull request merged**